### ISSUE TITLE
False positive: spec in quoted

### ISSUE THREAD

**Issue by hissssst** [rrrene/credo#1123]
### Environment

* Credo version (`mix credo -v`): `1.7.5-ref.master.c5b4d38+uncommittedchanges`
* Erlang/Elixir version (`elixir -v`):
```elixir
Erlang/OTP 25 [erts-13.2.2.6] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [jit:ns]

Elixir 1.15.7 (compiled with Erlang/OTP 25)
```
* Operating system: linux

### What were you trying to do?

I have a function:
```elixir
1  @spec to_def(t(), atom()) :: Macro.t()
2  def to_def(%__MODULE__{vars: vars, code: code}, name) do
3    quote generated: true do
4      def unquote(name)(unquote_splicing(vars)) do
5        unquote(code)
6      end
7    end
8  end
```

### Expected outcome

It does not complain that line 4 has no spec, since it is a generated function

### Actual outcome

It complains that the def on line 4 has no spec

```
  Code Readability
┃
┃ [R] → Functions should have a @spec type specification.
┃       lib/file.ex:4:column #(Module.to_def)
```

### Real world example

Run `mix credo` here: https://github.com/hissssst/pathex

**Comment by rrrene** [comment]
Thx for reporting! :+1:

**Comment by rrrene** [comment]
Thanks for reporting this 😀  It is now fixed on `master`.

You can try this by setting the Credo dep to

```elixir
{:credo, github: "rrrene/credo"}
```

Please report back if your issue is solved! 👍

**Comment by rrrene** [comment]
This is live in `v1.7.6` 🎉
